Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -119,6 +118,17 @@ private void CleanExpiredSessions()
             }
         }
 
+        internal void ValidateSecurePassword(SecureString password)
+        {
+            if (!ExtractPassword(Password).Equals(ExtractPassword(password)))
+            {
+                throw new Exception("Could not get a pool because of password mismatch");
```

Comment:
can you please add logging of such a case

Reply:
done